client/asset/eth/eth.go

```diff
@@ -304,21 +307,48 @@ func (eth *ExchangeWallet) MaxOrder(lotSize uint64, feeSuggestion uint64, nfo *d
 	if err != nil {
 		return nil, err
 	}
-	now := time.Now().Unix()
-	var secretHash [32]byte
-	copy(secretHash[:], encode.RandomBytes(32))
-	initGas, err := eth.node.initiateGas(eth.ctx, now, secretHash, &eth.acct.Address, &mainnetContractAddr)
+	initGas, err := eth.getInitGas()
 	if err != nil {
 		eth.log.Warnf("error getting init gas, falling back to server's value: %v", err)
 		initGas = nfo.SwapSize
 	}
 	availableBalance := balance.Available
 	maxTxFee := initGas * nfo.MaxFeeRate
-	realisticTxFee := initGas * feeSuggestion
 	lots := availableBalance / (lotSize + maxTxFee)
 	if lots < 1 {
 		return &asset.SwapEstimate{}, nil
 	}
+	est := eth.estimateSwap(lots, lotSize, feeSuggestion, nfo)
+	return est, nil
+}
+
+// PreSwap gets order estimates based on the available funds and the wallet
+// configuration.
+func (eth *ExchangeWallet) PreSwap(req *asset.PreSwapForm) (*asset.PreSwap, error) {
+	maxEst, err := eth.MaxOrder(req.LotSize, req.FeeSuggestion, req.AssetConfig)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
`PreSwap` only uses `maxEst.Lots`, which is fine I guess, but does do extra calculations. It looks like you could stick the lots check in estimateSwap pretty easily. It gets the InitGas anyway.